crates/iceberg/src/transaction.rs

```diff
@@ -169,6 +175,172 @@ impl<'a> Transaction<'a> {
 
         catalog.update_table(table_commit).await
     }
+
+    /// Adds existing parquet files
+    pub async fn add_parquet_files(
+        self,
+        file_paths: Vec<String>,
+        check_duplicate_files: bool,
+    ) -> Result<Transaction<'a>> {
+        if check_duplicate_files {
```

Comment:
This is different than in PyIceberg where we check if the file is already part of the table. Here we just check if duplicate entries are in the `file_paths` vector, which I don't think is very valuable.

If you add a file to a table, it is not easy to drop it without going into the actual metadata, therefore this check has been added.